spacer/task_utils.py

```diff
@@ -61,102 +65,316 @@ def check_extract_inputs(image: Image,
                 f" valid range of 0-{im_width - 1}.")
 
 
+class SplitMode(Enum):
+    """
+    How to split annotations between train, ref, and val sets.
+    """
+    # Each feature vector's points all go into train, or all go into ref, or
```

Comment:
Which part(s)? I have to say I was really getting hand-wavy on the 'more different' part, since I don't have a great grasp on the rationale myself. I'll see if I can think of a better way to word that part.

Reply:
Tried making the VECTORS comments more concrete and detailed.